pkg/dao/rpm_test.go

```diff
@@ -437,55 +438,48 @@ func (s *RpmSuite) TestInsertForRepositoryScenarioOverThreshold() {
 	s.genericInsertForRepository(testCases[scenarioOverThreshold])
 }
 
+func repoRpmCount(db *gorm.DB, repoUuid string) (int64, error) {
+	var rpmCount int64
+	err := db.
+		Table("rpms").
+		Joins("inner join repositories_rpms on repositories_rpms.rpm_uuid = rpms.uuid").
+		Where("repositories_rpms.repository_uuid = ?", repoUuid).
+		Count(&rpmCount).
+		Error
+	return rpmCount, err
+}
 func (s *RpmSuite) TestInsertForRepositoryWithExistingChecksums() {
 	t := s.Suite.T()
 	tx := s.tx
 	var rpm_count int64
 
 	pagedRpmInsertsLimit := 10
+	groupCount := 5
 
 	dao := GetRpmDao(tx, &RpmDaoOptions{
 		PagedRpmInsertsLimit: pointy.Int(pagedRpmInsertsLimit),
 	})
 	p := s.prepareScenarioRpms(scenarioThreshold, pagedRpmInsertsLimit)
-	records, err := dao.InsertForRepository(s.repo.Base.UUID, p[0:(pagedRpmInsertsLimit>>1)])
+	records, err := dao.InsertForRepository(s.repo.Base.UUID, p[0:groupCount])
```

Comment:
```suggestion
	err := deleteYumPackages(tx, p)
	require.NoError(t, err)

	records, err := dao.InsertForRepository(s.repo.Base.UUID, p[0:groupCount])
```

with a similar helper to the below:

```golang
func deleteYumPackages(db *gorm.DB, p []yum.Package) error {
	if db == nil {
		return fmt.Errorf("db cannot be nil")
	}
	var checksums []string = []string{}
	for _, pkg := range p {
		checksums = append(checksums, pkg.Checksum.Value)
	}
	return db.Unscoped().
		Where("checksum in (?)", checksums).
		Delete(&models.Rpm{}).
		Error
}
```

So it makes the test more predictable (very unlikely some match with current rpm records by the way).

Reply:
> So it makes the test more predictable (very unlikely some match with current rpm records by the way).

But here its based on a single repo, so counts should always be the same.  Unless i'm misunderstanding you?

Reply:
I don't recall it correctly!! I think my worry was related with the collision of the checksum during the test, so that it could return a different number, but I am not checking duplicity when preparing the data either at `prepareScenarioRpms`; and it is very unlikely.

Rethinking this again, I think it is not worthy the change I suggested; sorry for the noise.